bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -202,8 +320,11 @@ public void onNewDevice(Device device) {
 
     @Override
     public void onError() {
-        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Connection lost");
+        Duration delay = REINITIALIZE_DELAY_SECONDS;
+        synchronized (reInitializationCodeLock) {
+            scheduleReinitialize(delay);
```

Comment:
`scheduleReinitialize(REINITIALIZE_DELAY_SECONDS)`?

Reply:
Good point. I will look into that.

Reply:
I could not do that because the scheduler task is processed in two different methods -- `dispose()` resp. `scheduleReinitialize()`

Reply:
How does moving the synchronization to `scheduleReinitialize` affect the synchronization in `dispose()`?